Cython/Compiler/PyrexTypes.py

```diff
@@ -4380,14 +4405,47 @@ def specialize(self, values):
 
     def create_type_wrapper(self, env):
         from .UtilityCode import CythonUtilityCode
+        # Generate "int"-like conversion function
+        old_to_py_function = self.to_py_function
+        self.to_py_function = None
+        CIntLike.create_to_py_utility_code(self, env)
+        enum_to_pyint_func = self.to_py_function
+        self.to_py_function = old_to_py_function  # we don't actually want to overwrite this
+
         env.use_utility_code(CythonUtilityCode.load(
             "EnumType", "CpdefEnums.pyx",
             context={"name": self.name,
                      "items": tuple(self.values),
                      "enum_doc": self.doc,
+                     "enum_to_pyint_func": enum_to_pyint_func
                      },
             outer_module_scope=env.global_scope()))
 
+    def create_to_py_utility_code(self, env):
+        if self.to_py_function is not None:
+            return self.to_py_function
+        if self.entry.create_wrapper:
+            from .UtilityCode import CythonUtilityCode
+            self.to_py_function = "__Pyx_Enum_%s_to_py" % self.name
+            if self.entry.scope != env.global_scope():
+                module_name = self.entry.scope.qualified_name
+            else:
+                module_name = None
+            env.use_utility_code(CythonUtilityCode.load(
+                "EnumTypeToPy", "CpdefEnums.pyx",
+                context={"funcname": self.to_py_function,
+                        "name": self.name,
+                        "items": tuple(self.values),
+                        "underlying_type": "int",
+                        "module_name": module_name,
+                        "is_flag": True,
+                        },
+            outer_module_scope=self.entry.scope  # ensure that "name" is findable
+            ))
+            return True
+        else:
+            return super(CEnumType, self).create_to_py_utility_code(env)
```

Comment:
Better return the super result `if not self.entry.create_wrapper`, and otherwise continue with the normal handling, rather than keeping the if-else open over the whole function body.